cinema/views.py

```diff
@@ -1,10 +1,20 @@
 from datetime import datetime
 
 from django.db.models import F, Count
-from rest_framework import viewsets
+from rest_framework import viewsets, mixins, generics
+from rest_framework.authentication import TokenAuthentication
 from rest_framework.pagination import PageNumberPagination
-
-from cinema.models import Genre, Actor, CinemaHall, Movie, MovieSession, Order
+from rest_framework.permissions import IsAuthenticated
+
+from cinema.models import (
+    Genre,
+    Actor,
+    CinemaHall,
+    Movie,
+    MovieSession,
+    Order
+)
+from cinema.permissions import (IsAdminOrIfAuthenticatedReadOnly, )
```

Comment:
There is no need to import a single class inside of `()`, you could import it directly. Follow this import approach in the future.